cmake/RippledCore.cmake

```diff
@@ -60,7 +111,19 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS
 )
 target_sources(xrpl.libxrpl PRIVATE ${sources})
 
+target_link_modules(xrpl PUBLIC
+  basics
+  beast
+  crypto
+  json
+  protocol
+  resource
+  server
+)
+
 target_include_directories(xrpl.libxrpl
+  PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
```

Comment:
I think this is probably not needed ?

Reply:
actually, upon some testing, this whole `target_include_directories` seem to be not needed.

Reply:
It's not needed now that all of the subdirectories in libxrpl are modules. It was necessary as I migrated them one-by-one, and it could be necessary in the future if someone adds a non-module subdirectory. Could be nice if it "just works" without tampering with the CMake. What do you think?

Reply:
Perhaps comment out and put a comment above, to explain why this is not necessary and in what condition it might become necessary again ?